examples/ecs/observers.rs

```diff
@@ -0,0 +1,199 @@
+//! Demonstrates how to observe life-cycle triggers as well as define custom ones.
+
+use bevy::{
+    prelude::*,
+    utils::{HashMap, HashSet},
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(Update, (draw_shapes, handle_click))
+        .run();
+}
+
+#[derive(Component)]
+struct Mine {
+    pos: Vec2,
+    size: f32,
+}
+
+#[derive(Trigger)]
+struct TriggerMines {
+    pos: Vec2,
+    radius: f32,
+}
+
+#[derive(Trigger)]
+struct Explode;
+
+fn setup(world: &mut World) {
+    world.spawn(Camera2dBundle::default());
+
+    let font = world
+        .resource::<AssetServer>()
+        .load("fonts/FiraMono-Medium.ttf");
+    world.spawn(TextBundle::from_section(
+        "Click on a \"Mine\" to trigger it.\n\
+            When it explodes it will trigger all overlapping mines.",
+        TextStyle {
+            font,
+            font_size: 24.,
+            color: Color::WHITE,
+        },
+    ));
+
+    // Pre-register all our components, resource and trigger types.
+    world.init_resource::<SpatialIndex>();
+    world.init_component::<Mine>();
+    world.init_trigger::<TriggerMines>();
+    world.init_trigger::<Explode>();
+
+    // Observers run when a certain trigger is fired, each trigger is represented by a type.
+    // This observer runs whenever `TriggerMines` is fired, observers run systems which can be defined as closures.
```

Comment:
Bikeshed, but I don't think "firing" is the correct terminology now that we're calling these triggers instead of events. Maybe a better verb would be "activate" or even just "trigger" for some usages